### PR TITLE
Add partial partition fault to test framework for both docker and VM.

### PR DESCRIPTION
**Pull request by WenbinZhu**
### Description:
Make both docker and VM capable of creating partial partition in testing framework.

### Why should this be merged: 
First step of adding new test scenarios  for the new fault detector.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Comment by WenbinZhu** [comment]
@zalokhan Can we discuss and come up with a basic test case? I can add it to this pr.

**Comment by WenbinZhu** [comment]
Added new scenario test cases for link failure.

File: it/src/main/java/org/corfudb/universe/node/server/CorfuServer.java [hunk 1]
```diff
@@ -15,12 +17,20 @@
...
     /**
-     * Disconnect a CorfuServer from the network
+     * Disconnect a CorfuServer from the cluster, which creates a complete partition.
      *
      * @throws NodeException thrown in case of unsuccessful disconnect.
      */
     void disconnect();
 
+    /**
+     * Disconnect a CorfuServer from a list of other servers, which creates a partial partition.
```
Comment (StrikingThirteen) [inline it/src/main/java/org/corfudb/universe/node/server/CorfuServer.java]: Can you mention that this is a symmetric disconnect?

File: it/src/main/java/org/corfudb/universe/util/IpTablesUtil.java [hunk 2]
```diff
@@ -21,8 +21,28 @@ public static String dropInput(String ipAddress) {
...
+        return new String[] {"iptables", "-A", "OUTPUT", "-d", ipAddress, "-j", "DROP"};
+    }
+
+    /**
+     * Recover the drop input rule
+     *
+     * @param ipAddress ip address to recover the drop input rule
+     * @return command line
+     */
+    public static String[] recoverDropInput(String ipAddress) {
```
Comment (StrikingThirteen) [inline it/src/main/java/org/corfudb/universe/util/IpTablesUtil.java]: recover is a little confusing. We are deleting a rule added before. maybe deleteDropInput or revertDropInputRule?

File: it/src/main/java/org/corfudb/universe/node/server/CorfuServer.java (hunk 1 above)
Comment (WenbinZhu) [reply it/src/main/java/org/corfudb/universe/node/server/CorfuServer.java]: Done.

File: it/src/main/java/org/corfudb/universe/util/IpTablesUtil.java (hunk 2 above)
Comment (WenbinZhu) [reply it/src/main/java/org/corfudb/universe/util/IpTablesUtil.java]: Done.

**Pull request merged**